### PR TITLE
core: disable color-name-sanitization

### LINKED ISSUE #1823: Rename color text always getting set to all lowercase without spaces
### What happened?

When I try to right click on a specific color, purple in my example, and then click "Rename color" and type my desired name (for example, Official Notes) and click on Save. 

It ignores any formatting, becomes case insensitive, ignores spaces and gets saved as "officialnotes". I tested and it appears to accept special characters but not spaces.

### Steps to reproduce the problem

1. Right click on any color from left pane
2. Click "Rename color"
3. Type any desired name (with a mix of uppercase, lowercase and spaces)
4. Click on Save
5. You will see the effect

### Version

v2.4.0

### Platform/OS

Windows

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by Abdulrehman-Jafer**
Fixes #1823

### CODE REVIEW

File: packages/core/collections/tags.js [hunk 1]
```diff
@@ -72,15 +72,19 @@ export default class Tags extends Collection {
...
     }
+    let sanitizedName = this.sanitize(newName);
+    if (!sanitizedName)
+      throw new Error(`${isColorTag ? "Color" : "Tag"} title cannot be empty.`);
 
-    newName = this.sanitize(newName);
-    if (!newName) throw new Error("Tag title cannot be empty.");
+    if (!isColorTag) {
+      newName = sanitizedName;
+    }
```
Comment (thecodrr) [inline packages/core/collections/tags.js:87]: Just don't do any sanitization on rename. There's no need to introduce a new parameter.

Comment (Abdulrehman-Jafer) [reply packages/core/collections/tags.js:87]: Sanitization is also required for renaming color as if we don't the user can save a color name with only spaces like "        " which will not be good.

Comment (thecodrr) [reply packages/core/collections/tags.js:87]: > which will not be good

Yeah I don't think that's an issue. If a user does that they can rename again. Renaming should allow full freedom.

Comment (Abdulrehman-Jafer) [reply packages/core/collections/tags.js:87]: Ok

File: packages/core/collections/tags.js [hunk 2]
```diff
@@ -72,15 +72,15 @@ export default class Tags extends Collection {
...
+  async rename(tagId, newName, isColorTag) {
     let tag = this.tag(tagId);
     if (!tag) {
       console.error(`No tag found. Tag id:`, tagId);
       return;
     }
 
-    newName = this.sanitize(newName);
-    if (!newName) throw new Error("Tag title cannot be empty.");
+    if (!isColorTag) newName = this.sanitize(newName);
```
Comment (thecodrr) [inline packages/core/collections/tags.js:82]: There is no need to do any sanitization in any case during renaming.

File: apps/web/src/common/dialog-controller.tsx [hunk 3]
```diff
@@ -209,9 +209,8 @@ export function showError(title: string, message: string) {
 export function showMultiDeleteConfirmation(length: number) {
   return confirm({
     title: `Delete ${length} items?`,
-    message: `These items will be **kept in your Trash for ${
-      db.settings?.getTrashCleanupInterval() || 7
-    } days** after which they will be permanently deleted.`,
+    message: `These items will be **kept in your Trash for ${db.settings?.getTrashCleanupInterval() || 7
+      } days** after which they will be permanently deleted.`,
```
Comment (thecodrr) [inline apps/web/src/common/dialog-controller.tsx:213]: What's this change?

Comment (Abdulrehman-Jafer) [reply apps/web/src/common/dialog-controller.tsx:213]: Just auto formatted by prettier.

File: packages/core/collections/tags.js (hunk 2 above)
Comment (Abdulrehman-Jafer) [reply packages/core/collections/tags.js:82]: Ok

**Comment by Abdulrehman-Jafer** [comment]
Now the sanitization is disabled for both the tags and colors.

**Review by thecodrr — APPROVED** [review]

**Pull request merged**